controllers/permission_controller.go

```diff
@@ -44,13 +44,31 @@ func (r *PermissionReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return reconcile.Result{}, client.IgnoreNotFound(err)
 	}
 
-	rabbitClient, err := r.RabbitmqClientFactory(ctx, r.Client, permission.Spec.RabbitmqClusterReference, permission.Namespace)
-	// If the object is not being deleted, but the RabbitmqCluster no longer exists, it could be that
-	// the Cluster is temporarily down. Requeue until it comes back up.
-	if errors.Is(err, internal.NoSuchRabbitmqClusterError) && permission.ObjectMeta.DeletionTimestamp.IsZero() {
+	systemCertPool, err := x509.SystemCertPool()
+	if err != nil {
+		msg := "failed to retrieve system trusted certs"
+		r.Recorder.Event(permission, corev1.EventTypeWarning, "FailedUpdate", msg)
+		logger.Error(err, msg)
+		return ctrl.Result{}, err
+	}
```

Comment:
These lines show up in every controller. The logic here can be extract to a helper "extractSystemCertPool". Obviously you still need to do the error check, however these 3 lines could be omitted:
```
msg := "failed to retrieve system trusted certs"
r.Recorder.Event(q, corev1.EventTypeWarning, "FailedUpdate", msg)
logger.Error(err, msg)
```
What do you think?